evap/evaluation/tools.py

```diff
@@ -33,6 +34,15 @@
 CV = TypeVar("CV", bound=CellValue)
 
 
+@contextmanager
+def temporary_receiver(signal, func, **kwargs):
+    # semi-copy of https://github.com/django/django/blob/3266f2516c27dd25abebe8e8f7b8778650ab4f18/django/dispatch/dispatcher.py#L472
+    # with sane contextmanager behavior (receivers are unlinked on exit). We don't support lists of signals (for now).
+    signal.connect(receiver=func, **kwargs)
+    yield
+    signal.disconnect(receiver=func, **kwargs)
```

Comment:
Shouldn't we do the following?
```suggestion
    try:
        yield
    finally:
        signal.disconnect(receiver=func, **kwargs)
```

Reply:
absolutely correct